File: inject-java/src/test/groovy/io/micronaut/inject/annotation/MyAnnotation2.java
package io.micronaut.inject.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Documented
@Retention(RUNTIME)
@Target({ElementType.TYPE})
public @interface MyAnnotation2 {

    int num() default 10;

    String value() default "";

    boolean bool() default false;

    MyAnnotation3 ann() default @MyAnnotation3("foo");

    MyEnum2 myEnum() default MyEnum2.ABC;

    int[] intArray1() default {};

    int[] intArray2() default {1, 2, 3};

    int[] intArray3();

    String[] stringArray1() default {};

    String[] stringArray2() default {""};

    String[] stringArray3() default {"A"};

    String[] stringArray4();

    boolean[] boolArray1() default {};

    boolean[] boolArray2() default {true};

    boolean[] boolArray3() default {false};

    boolean[] boolArray4();

    MyEnum2[] myEnumArray1() default {};

    MyEnum2[] myEnumArray2() default {MyEnum2.ABC};

    MyEnum2[] myEnumArray3() default {MyEnum2.FOO, MyEnum2.BAR};

    MyEnum2[] myEnumArray4();

    Class[] classesArray1() default {};

    Class[] classesArray2() default {String.class};

    MyAnnotation3[] annotationsArray1() default {};

    MyAnnotation3[] annotationsArray2() default {@MyAnnotation3("foo"), @MyAnnotation3("bar")};

}

File: test-inject-kotlin2-ksp2/src/test/kotlin/io/micronaut/kotlin/processing/beans/MyAnnotation2.kt
package io.micronaut.kotlin.processing.beans

import kotlin.reflect.KClass

@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class MyAnnotation2(
    val num: Int = 10,
    val value: String = "",
    val bool: Boolean = false,
    val ann: MyAnnotation3 = MyAnnotation3("foo"),
    val myEnum: MyEnum2 = MyEnum2.ABC,
    val intArray1: IntArray = [],
    val intArray2: IntArray = [1, 2, 3],
    val intArray3: IntArray,
    val stringArray1: Array<String> = [],
    val stringArray2: Array<String> = [""],
    val stringArray3: Array<String> = ["A"],
    val stringArray4: Array<String>,
    val boolArray1: BooleanArray = [],
    val boolArray2: BooleanArray = [true],
    val boolArray3: BooleanArray = [false],
    val boolArray4: BooleanArray,
    val myEnumArray1: Array<MyEnum2> = [],
    val myEnumArray2: Array<MyEnum2> = [MyEnum2.ABC],
    val myEnumArray3: Array<MyEnum2> = [MyEnum2.FOO, MyEnum2.BAR],
    val myEnumArray4: Array<MyEnum2>,
    val classesArray1: Array<KClass<*>> = [],
    val classesArray2: Array<KClass<*>> = [String::class],
    val annotationsArray1: Array<MyAnnotation3> = [],
    val annotationsArray2: Array<MyAnnotation3> = [MyAnnotation3(
        "foo"
    ), MyAnnotation3("bar")]
)


@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class MyAnnotation3(val value: String = "")

enum class MyEnum2 {
    FOO, BAR, ABC
}

